Rollup of 17 pull requests

kennytm opened the pull request:
Successful merges:

 - #53030 (Updated RELEASES.md for 1.29.0)
 - #53104 (expand the documentation on the `Unpin` trait)
 - #53213 (Stabilize IP associated constants)
 - #53296 (When closure with no arguments was expected, suggest wrapping)
 - #53329 (Replace usages of ptr::offset with ptr::{add,sub}.)
 - #53363 (add individual docs to `core::num::NonZero*`)
 - #53370 (Stabilize macro_vis_matcher)
 - #53393 (Mark libserialize functions as inline)
 - #53405 (restore the page title after escaping out of a search)
 - #53452 (Change target triple used to check for lldb in build-manifest)
 - #53462 (Document Box::into_raw returns non-null ptr)
 - #53465 (Remove LinkMeta struct)
 - #53492 (update lld submodule to include RISCV patch)
 - #53496 (Fix typos found by codespell.)
 - #53521 (syntax: Optimize some literal parsing)
 - #53540 (Moved issue-53157.rs into src/test/ui/consts/const-eval/)
 - #53551 (Avoid some Place clones.)


Failed merges:


r? @ghost

bors commented:
:hourglass: Testing commit 76a760ccecc0d6788522540b46c07812365baae0 with merge a95e26dc2447cbfcb1ec52abb4152f21f43816d1...

rust-highfive commented:
The job `x86_64-gnu-llvm-5.0` of your PR [failed on Travis](https://travis-ci.org/rust-lang/rust/jobs/418333109) ([raw log](https://api.travis-ci.org/v3/job/418333109/log.txt)). Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.

<details><summary><i>Click to expand the log.</i></summary>

```plain
[00:02:09] Successfully tagged rust-ci:latest
[00:02:09] Built container sha256:62a414b53655fac5e0afb30fcd2057090b2dc68ffc6cb2cc2732b30823cf2999
[00:02:09] Uploading finished image to s3://rust-lang-ci-sccache2/docker/a08f738800c5d4bf390acb73ec0118b6ff1146863e6648bef20b9a0326bcba9bd46d442bc30b69fac421c987dce13ae565bdfc6c2fccdb7c32851f08d351168d
[00:02:09] 
[00:02:09] Partial credentials found in env, missing: AWS_SECRET_ACCESS_KEY
[00:02:14] xargs: docker: terminated by signal 13

[00:02:14] travis_time:end:219eb154:start=1534787385211093754,finish=1534787453156524923,duration=67945431169
[CI_JOB_NAME=x86_64-gnu-llvm-5.0]
[00:02:14] [CI_JOB_NAME=x86_64-gnu-llvm-5.0]
---
[00:45:08] ....................................................................................................
[00:45:11] ....................................................................................................
[00:45:14] ....................................................................................................
[00:45:17] .............i......................................................................................
[00:45:21] .......................................................................................F...........i
[00:45:27] .............ii.iii.................................................................................
[00:45:30] ....................................................................................................
[00:45:32] ....................................................................................................
[00:45:34] ....................................................................................................
---
[00:46:58] ---- [ui] ui/consts/const-size_of-cycle.rs stdout ----
[00:46:58] diff of stderr:
[00:46:58] 
[00:46:58] 2    |
[00:46:58] 3 note: ...which requires normalizing `ParamEnvAnd { param_env: ParamEnv { caller_bounds: [], reveal: All }, value: [u8; _] }`...
[00:46:58] 4 note: ...which requires const-evaluating `Foo::bytes::{{constant}}`...
[00:46:58] -   --> $SRC_DIR/libcore/mem.rs:323:14
[00:46:58] +   --> $SRC_DIR/libcore/mem.rs:291:14
[00:46:58] 6    |
[00:46:58] 7 LL |     unsafe { intrinsics::size_of::<T>() }
[00:46:58] 
[00:46:58] 
[00:46:58] 9    = note: ...which again requires computing layout of `Foo`, completing the cycle
[00:46:58] 10 note: cycle used when const-evaluating `Foo::bytes::{{constant}}`
[00:46:58] -   --> $SRC_DIR/libcore/mem.rs:323:14
[00:46:58] +   --> $SRC_DIR/libcore/mem.rs:291:14
[00:46:58] 12    |
[00:46:58] 13 LL |     unsafe { intrinsics::size_of::<T>() }
[00:46:58] 
[00:46:58] 
[00:46:58] The actual stderr differed from the expected stderr.
[00:46:58] Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/consts/const-size_of-cycle/const-size_of-cycle.stderr
[00:46:58] Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/consts/const-size_of-cycle/const-size_of-cycle.stderr
[00:46:58] To update references, rerun the tests and pass the `--bless` flag
[00:46:58] To only update this specific test, also pass `--test-args consts/const-size_of-cycle.rs`
[00:46:58] error: 1 errors occurred comparing output.
[00:46:58] status: exit code: 1
[00:46:58] status: exit code: 1
[00:46:58] command: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/consts/const-size_of-cycle.rs" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "-Zui-testing" "-C" "prefer-dynamic" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/consts/const-size_of-cycle/a" "-Crpath" "-O" "-Zunstable-options" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/consts/const-size_of-cycle/auxiliary" "-A" "unused"
[00:46:58] ------------------------------------------
[00:46:58] 
[00:46:58] ------------------------------------------
[00:46:58] stderr:
[00:46:58] stderr:
[00:46:58] ------------------------------------------
[00:46:58] {"message":"cycle detected when computing layout of `Foo`","code":{"code":"E0391","explanation":"\nThis error indicates that some types or traits depend on each other\nand therefore cannot be constructed.\n\nThe following example contains a circular dependency between two traits:\n\n```compile_fail,E0391\ntrait FirstTrait : SecondTrait {\n\n}\n\ntrait SecondTrait : FirstTrait {\n\n}\n```\n"},"level":"error","spans":[{"file_name":"/checkout/src/test/ui/consts/const-size_of-cycle.rs","byte_start":0,"byte_end":0,"line_start":1,"line_end":1,"column_start":1,"column_end":1,"is_primary":true,"text":[{"text":"// Copyright 2017 The Rust Project Developers. See the COPYRIGHT","highlight_start":1,"highlight_end":1}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"...which requires normalizing `ParamEnvAnd { param_env: ParamEnv { caller_bounds: [], reveal: All }, value: [u8; _] }`...","code":null,"level":"note","spans":[{"file_name":"/checkout/src/test/ui/consts/const-size_of-cycle.rs","byte_start":0,"byte_end":0,"line_start":1,"line_end":1,"column_start":1,"column_end":1,"is_primary":true,"text":[{"text":"// Copyright 2017 The Rust Project Developers. See the COPYRIGHT","highlight_start":1,"highlight_end":1}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":null},{"message":"...which requires const-evaluating `Foo::bytes::{{constant}}`...","code":null,"level":"note","spans":[{"file_name":"/checkout/src/libcore/mem.rs","byte_start":9648,"byte_end":9674,"line_start":291,"line_end":291,"column_start":14,"column_end":40,"is_primary":true,"text":[{"text":"    unsafe { intrinsics::size_of::<T>() }","highlight_start":14,"highlight_end":40}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":null},{"message":"...which again requires computing layout of `Foo`, completing the cycle","code":null,"level":"note","spans":[],"children":[],"rendered":null},{"message":"cycle used when const-evaluating `Foo::bytes::{{constant}}`","code":null,"level":"note","spans":[{"file_name":"/checkout/src/libcore/mem.rs","byte_start":9648,"byte_end":9674,"line_start":291,"line_end":291,"column_start":14,"column_end":40,"is_primary":true,"text":[{"text":"    unsafe { intrinsics::size_of::<T>() }","highlight_start":14,"highlight_end":40}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":null}],"rendered":"error[E0391]: cycle detected when computing layout of `Foo`\n   |\nnote: ...which requires normalizing `ParamEnvAnd { param_env: ParamEnv { caller_bounds: [], reveal: All }, value: [u8; _] }`...\nnote: ...which requires const-evaluating `Foo::bytes::{{constant}}`...\n  --> /checkout/src/libcore/mem.rs:291:14\n   |\nLL |     unsafe { intrinsics::size_of::<T>() }\n   |              ^^^^^^^^^^^^^^^^^^^^^^^^^^\n   = note: ...which again requires computing layout of `Foo`, completing the cycle\nnote: cycle used when const-evaluating `Foo::bytes::{{constant}}`\n  --> /checkout/src/libcore/mem.rs:291:14\n   |\nLL |     unsafe { intrinsics::size_of::<T>() }\n   |              ^^^^^^^^^^^^^^^^^^^^^^^^^^\n\n"}
[00:46:58] {"message":"aborting due to previous error","code":null,"level":"error","spans":[],"children":[],"rendered":"error: aborting due to previous error\n\n"}
[00:46:58] {"message":"For more information about this error, try `rustc --explain E0391`.","code":null,"level":"","spans":[],"children":[],"rendered":"For more information about this error, try `rustc --explain E0391`.\n"}
[00:46:58] ------------------------------------------
[00:46:58] 
[00:46:58] thread '[ui] ui/consts/const-size_of-cycle.rs' panicked at 'explicit panic', tools/compiletest/src/runtest.rs:3189:9
[00:46:58] note: Run with `RUST_BACKTRACE=1` for a backtrace.
---
[00:46:58] 
[00:46:58] thread 'main' panicked at 'Some tests failed', tools/compiletest/src/main.rs:497:22
[00:46:58] 
[00:46:58] 
[00:46:58] command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-tools-bin/compiletest" "--compile-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib" "--run-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-linux-gnu/lib" "--rustc-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "--src-base" "/checkout/src/test/ui" "--build-base" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui" "--stage-id" "stage2-x86_64-unknown-linux-gnu" "--mode" "ui" "--target" "x86_64-unknown-linux-gnu" "--host" "x86_64-unknown-linux-gnu" "--llvm-filecheck" "/usr/lib/llvm-5.0/bin/FileCheck" "--host-rustcflags" "-Crpath -O -Zunstable-options " "--target-rustcflags" "-Crpath -O -Zunstable-options  -Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--docck-python" "/usr/bin/python2.7" "--lldb-python" "/usr/bin/python2.7" "--gdb" "/usr/bin/gdb" "--quiet"Mon, 20 Aug 2018 18:35:37 GMT

The command "date && (curl -fs --head https://google.com | grep ^Date: | sed 's/Date: //g' || true)
" exited with 0.
travis_fold:start:after_failure.1
```

</details><p></p>

[I'm a bot](https://github.com/rust-ops/rust-log-analyzer)! I can only do what humans tell me to, so if this was not helpful or you have suggestions for improvements, please ping or otherwise contact **`@TimNN`**. ([Feature Requests](https://github.com/rust-ops/rust-log-analyzer/issues?q=is%3Aopen+is%3Aissue+label%3Afeature-request))

kennytm commented:
@bors r-

kennytm commented:
@bors r+

Removed #53503 (somehow the change affected existing UI tests 🤔)

Added #53521, #53363, #53462

bors commented:
:pushpin: Commit fd0a6d278e71a0c05c5caa5e1b8ee61c84d1ed6b has been approved by `kennytm`

<!-- @bors r=kennytm fd0a6d278e71a0c05c5caa5e1b8ee61c84d1ed6b -->

bors commented:
:hourglass: Testing commit fd0a6d278e71a0c05c5caa5e1b8ee61c84d1ed6b with merge 2c61dba813a6ac96444b2309aba68afcd942c28f...

bors commented:
:broken_heart: Test failed - [status-travis](https://travis-ci.org/rust-lang/rust/builds/418534786?utm_source=github_status&utm_medium=notification)

rust-highfive commented:
Your PR [failed on Travis](https://travis-ci.org/rust-lang/rust/jobs/418534813) ([raw log](https://api.travis-ci.org/v3/job/418534813/log.txt)). Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.

<details><summary><i>Click to expand the log.</i></summary>

```plain
[01:02:35] Tue Aug 21 06:11:18 UTC 2018 - building ...
[01:03:05] Tue Aug 21 06:11:48 UTC 2018 - building ...
[01:03:35] Tue Aug 21 06:12:18 UTC 2018 - building ...

Broadcast message from root@travis-job-bd09582f-0c77-46ee-a669-6dbee014ccdf
 (unknown) at 6:12 ...
The system is going down for power off NOW!
```

</details><p></p>

[I'm a bot](https://github.com/rust-ops/rust-log-analyzer)! I can only do what humans tell me to, so if this was not helpful or you have suggestions for improvements, please ping or otherwise contact **`@TimNN`**. ([Feature Requests](https://github.com/rust-ops/rust-log-analyzer/issues?q=is%3Aopen+is%3Aissue+label%3Afeature-request))

kennytm commented:
@bors retry travis-ci/travis-ci#4924

bors commented:
:hourglass: Testing commit fd0a6d278e71a0c05c5caa5e1b8ee61c84d1ed6b with merge 92ded590c109b9b3349be815c8994107301c9591...

bors commented:
:broken_heart: Test failed - [status-appveyor](https://ci.appveyor.com/project/rust-lang/rust/build/1.0.8688)

rust-highfive commented:
The job `dist-various-2` of your PR [failed on Travis](https://travis-ci.org/rust-lang/rust/jobs/418594184) ([raw log](https://api.travis-ci.org/v3/job/418594184/log.txt)). Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.

<details><summary><i>Click to expand the log.</i></summary>

```plain
[00:46:32]    Compiling panic_unwind v0.0.0 (file:///checkout/src/libpanic_unwind)
[00:46:33] [RUSTC-TIMING] panic_unwind test:false 0.252
[00:46:39] warning: redundant linker flag specified for library `zircon`
[00:46:39] 
[00:46:47] error: linking with `x86_64-fuchsia-clang` failed: exit code: 1
[00:46:47]   |
[00:46:47]   = note: "x86_64-fuchsia-clang" "-m64" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-fuchsia/lib" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-std/x86_64-fuchsia/release/deps/std-c4134ff9a92b454e.std.82r4c1p9-cgu.0.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-std/x86_64-fuchsia/release/deps/std-c4134ff9a92b454e.std.82r4c1p9-cgu.1.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-std/x86_64-fuchsia/release/deps/std-c4134ff9a92b454e.std.82r4c1p9-cgu.10.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-std/x86_64-fuchsia/release/deps/std-c4134ff9a92b454e.std.82r4c1p9-cgu.11.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-std/x86_64-fuchsia/release/deps/std-c4134ff9a92b454e.std.82r4c1p9-cgu.12.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-std/x86_64-fuchsia/release/deps/std-c4134ff9a92b454e.std.82r4c1p9-cgu.13.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-std/x86_64-fuchsia/release/deps/std-c4134ff9a92b454e.std.82r4c1p9-cgu.14.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-std/x86_64-fuchsia/release/deps/std-c4134ff9a92b454e.std.82r4c1p9-cgu.15.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-std/x86_64-fuchsia/release/deps/std-c4134ff9a92b454e.std.82r4c1p9-cgu.2.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-std/x86_64-fuchsia/release/deps/std-c4134ff9a92b454e.std.82r4c1p9-cgu.3.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-std/x86_64-fuchsia/release/deps/std-c4134ff9a92b454e.std.82r4c1p9-cgu.4.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-std/x86_64-fuchsia/release/deps/std-c4134ff9a92b454e.std.82r4c1p9-cgu.5.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-std/x86_64-fuchsia/release/deps/std-c4134ff9a92b454e.std.82r4c1p9-cgu.6.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-std/x86_64-fuchsia/release/deps/std-c4134ff9a92b454e.std.82r4c1p9-cgu.7.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-std/x86_64-fuchsia/release/deps/std-c4134ff9a92b454e.std.82r4c1p9-cgu.8.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-std/x86_64-fuchsia/release/deps/std-c4134ff9a92b454e.std.82r4c1p9-cgu.9.rcgu.o" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-std/x86_64-fuchsia/release/deps/libstd-c4134ff9a92b454e.so" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-std/x86_64-fuchsia/release/deps/std-c4134ff9a92b454e.4ichl4vie5umi5bw.rcgu.o" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-std/x86_64-fuchsia/release/deps/std-c4134ff9a92b454e.50idbkajsw1eibsc.rcgu.o" "-Wl,-O1" "-nodefaultlibs" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-std/x86_64-fuchsia/release/deps" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-std/release/deps" "-L" "/checkout/obj/build/x86_64-fuchsia/native/libbacktrace/" "-L" "/checkout/obj/build/x86_64-fuchsia/native/libbacktrace" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-std/x86_64-fuchsia/release/build/compiler_builtins-6824f47f17cd9867/out" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-fuchsia/lib" "-lzircon" "-Wl,-Bstatic" "-Wl,--whole-archive" "-lbacktrace" "-Wl,--no-whole-archive" "-Wl,-Bdynamic" "-lfdio" "-Wl,-Bstatic" "-Wl,--whole-archive" "/tmp/rustcHshpBO/libpanic_unwind-4e90336c2dc63069.rlib" "-Wl,--no-whole-archive" "-Wl,--whole-archive" "/tmp/rustcHshpBO/libunwind-dd1bd39b57b8e229.rlib" "-Wl,--no-whole-archive" "-Wl,--whole-archive" "/tmp/rustcHshpBO/liballoc_system-5794f65df23a065e.rlib" "-Wl,--no-whole-archive" "-Wl,--whole-archive" "/tmp/rustcHshpBO/liblibc-3955de38374473d4.rlib" "-Wl,--no-whole-archive" "-Wl,--whole-archive" "/tmp/rustcHshpBO/liballoc-2b46423f96c165ca.rlib" "-Wl,--no-whole-archive" "-Wl,--whole-archive" "/tmp/rustcHshpBO/libcore-ded92bdb6f611f3a.rlib" "-Wl,--no-whole-archive" "/tmp/rustcHshpBO/libcompiler_builtins-e6fbe755c60fe6f9.rlib" "-Wl,-Bdynamic" "-lunwind" "-lc" "-lfdio" "-shared" "-Wl,-rpath,$ORIGIN/../lib"
[00:46:47]   = note: ld.lld: error: unable to find library -lzircon
[00:46:47]           ld.lld: error: unable to find library -lfdio
[00:46:47]           ld.lld: error: unable to find library -lc
[00:46:47]           ld.lld: error: unable to find library -lfdio
[00:46:47]           clang-8: error: ld.lld command failed with exit code 1 (use -v to see invocation)
[00:46:47] 
[00:46:47] error: aborting due to previous error
[00:46:47] 
[00:46:47] [RUSTC-TIMING] std test:false 14.776
[00:46:47] [RUSTC-TIMING] std test:false 14.776
[00:46:47] error: Could not compile `std`.
[00:46:47] 
[00:46:47] Caused by:
[00:46:47]   process didn't exit successfully: `/checkout/obj/build/bootstrap/debug/rustc --crate-name std libstd/lib.rs --color always --error-format json --crate-type dylib --crate-type rlib --emit=dep-info,link -C prefer-dynamic -C opt-level=2 --cfg 'feature="alloc_jemalloc"' --cfg 'feature="backtrace"' --cfg 'feature="jemalloc"' --cfg 'feature="panic-unwind"' --cfg 'feature="panic_unwind"' -C metadata=c4134ff9a92b454e -C extra-filename=-c4134ff9a92b454e --out-dir /checkout/obj/build/x86_64-unknown-linux-gnu/stage2-std/x86_64-fuchsia/release/deps --target x86_64-fuchsia -L dependency=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-std/x86_64-fuchsia/release/deps -L dependency=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-std/release/deps --extern alloc=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-std/x86_64-fuchsia/release/deps/liballoc-2b46423f96c165ca.rlib --extern alloc_jemalloc=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-std/x86_64-fuchsia/release/deps/liballoc_jemalloc-24599c07847121d9.rlib --extern alloc_system=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-std/x86_64-fuchsia/release/deps/liballoc_system-5794f65df23a065e.rlib --extern compiler_builtins=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-std/x86_64-fuchsia/release/deps/libcompiler_builtins-e6fbe755c60fe6f9.rlib --extern core=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-std/x86_64-fuchsia/release/deps/libcore-ded92bdb6f611f3a.rlib --extern libc=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-std/x86_64-fuchsia/release/deps/liblibc-3955de38374473d4.rlib --extern panic_abort=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-std/x86_64-fuchsia/release/deps/libpanic_abort-d4a3131cbb1d98ab.rlib --extern panic_unwind=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-std/x86_64-fuchsia/release/deps/libpanic_unwind-4e90336c2dc63069.rlib --extern unwind=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-std/x86_64-fuchsia/release/deps/libunwind-dd1bd39b57b8e229.rlib -L native=/checkout/obj/build/x86_64-fuchsia/native/libbacktrace/ -L native=/checkout/obj/build/x86_64-fuchsia/native/libbacktrace -l static=backtrace -l static=backtrace -l zircon -l fdio -L native=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-std/x86_64-fuchsia/release/build/compiler_builtins-6824f47f17cd9867/out` (exit code: 1)
[00:46:47] command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo" "build" "--target" "x86_64-fuchsia" "-j" "4" "--release" "--locked" "--color" "always" "--features" "panic-unwind jemalloc backtrace" "--manifest-path" "/checkout/src/libstd/Cargo.toml" "--message-format" "json"
[00:46:47] expected success, got: exit code: 101
[00:46:47] thread 'main' panicked at 'cargo must succeed', bootstrap/compile.rs:1119:9
[00:46:47] travis_fold:end:stage2-std

[00:46:47] travis_time:end:stage2-std:start=1534844033650996955,finish=1534844089051173264,duration=55400176309

---
travis_time:end:17fae05e:start=1534844090331571055,finish=1534844090339958685,duration=8387630
travis_fold:end:after_failure.3
travis_fold:start:after_failure.4
travis_time:start:0d47f4bb
$ ln -s . checkout && for CORE in obj/cores/core.*; do EXE=$(echo $CORE | sed 's|obj/cores/core\.[0-9]*\.!checkout!\(.*\)|\1|;y|!|/|'); if [ -f "$EXE" ]; then printf travis_fold":start:crashlog\n\033[31;1m%s\033[0m\n" "$CORE"; gdb -q -c "$CORE" "$EXE" -iex 'set auto-load off' -iex 'dir src/' -iex 'set sysroot .' -ex bt -ex q; echo travis_fold":"end:crashlog; fi; done || true
travis_fold:end:after_failure.4
travis_fold:start:after_failure.5
travis_time:start:1ac4d3aa
travis_time:start:1ac4d3aa
$ cat ./obj/build/x86_64-unknown-linux-gnu/native/asan/build/lib/asan/clang_rt.asan-dynamic-i386.vers || true
cat: ./obj/build/x86_64-unknown-linux-gnu/native/asan/build/lib/asan/clang_rt.asan-dynamic-i386.vers: No such file or directory
travis_fold:end:after_failure.5
travis_fold:start:after_failure.6
travis_time:start:263be1fc
$ dmesg | grep -i kill
```

</details><p></p>

[I'm a bot](https://github.com/rust-ops/rust-log-analyzer)! I can only do what humans tell me to, so if this was not helpful or you have suggestions for improvements, please ping or otherwise contact **`@TimNN`**. ([Feature Requests](https://github.com/rust-ops/rust-log-analyzer/issues?q=is%3Aopen+is%3Aissue+label%3Afeature-request))

kennytm commented:
@bors r+

Removed #53480 (apparent test failure)
Removed #53329 (already testing)

Added #53104, #53540.

bors commented:
:pushpin: Commit 9bbab65ebefbeb512445a759603b03f858e5e9b9 has been approved by `kennytm`

<!-- @bors r=kennytm 9bbab65ebefbeb512445a759603b03f858e5e9b9 -->

bors commented:
:hourglass: Testing commit 9bbab65ebefbeb512445a759603b03f858e5e9b9 with merge 21cc9397c766831c58b15866709f064944c504ad...

bors commented:
:broken_heart: Test failed - [status-travis](https://travis-ci.org/rust-lang/rust/builds/418640368?utm_source=github_status&utm_medium=notification)

rust-highfive commented:
The job `dist-x86_64-linux` of your PR [failed on Travis](https://travis-ci.org/rust-lang/rust/jobs/418640370) ([raw log](https://api.travis-ci.org/v3/job/418640370/log.txt)). Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.

<details><summary><i>Click to expand the log.</i></summary>

```plain
[00:01:38] 
[00:01:38] Total download size: 4.9 M
[00:01:38] Downloading Packages:
[00:01:38] --------------------------------------------------------------------------------
[00:01:38] Total                                            22 MB/s | 4.9 MB     00:00     
[00:01:38] warning: rpmts_HdrFromFdno: Header V3 DSA signature: NOKEY, key ID e8562897
[00:01:38] Importing GPG key 0xE8562897 "CentOS-5 Key (CentOS 5 Official Signing Key) <centos-5-key@centos.org>" from /etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-5
[00:01:38] Running Transaction Test
[00:01:38] Finished Transaction Test
[00:01:38] Transaction Test Succeeded
[00:01:38] Running Transaction
---
[00:03:13] + hide_output make install
[00:03:13] + set +x
[00:03:32] shared.sh: line 11:   351 Terminated              bash -c "while true; do sleep 30; echo \$(date) - building ...; done"
[00:03:32] + cd ..
[00:03:32] + rm -rf openssl-1.0.2k
[00:03:32] ./build-openssl.sh: line 25:  4114 Terminated              bash -c "while true; do sleep 30; echo \$(date) - building ...; done"  (wd: /tmp/openssl-1.0.2k)
[00:03:32] + ln -nsf /etc/pki/tls/cert.pem /rustroot/ssl/
[00:03:33] Removing intermediate container 86c11a83fc6f
[00:03:33] Step 14/38 : COPY dist-x86_64-linux/build-curl.sh /tmp/
[00:03:33]  ---> 447a409ad779
[00:03:33] Step 15/38 : RUN ./build-curl.sh
[00:03:33] Step 15/38 : RUN ./build-curl.sh
[00:03:33]  ---> Running in 03e88d8bcbcd
[00:03:33] + source shared.sh
[00:03:33] + VERSION=7.51.0
[00:03:33] + curl http://cool.haxx.se/download/curl-7.51.0.tar.bz2
[00:03:35]   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
[00:03:35]                                  Dload  Upload   Total   Spent    Left  Speed
[00:03:36] 
  0 2509k    0 14215    0     0   9165      0  0:04:40  0:00:01  0:04:39  9165
  0 2509k    0 14215    0     0   9165      0  0:04:40  0:00:01  0:04:39  9165
 52 2509k   52 1310k    0     0   533k      0  0:00:04  0:00:02  0:00:02 1426k
100 2509k  100 2509k    0     0   908k      0  0:00:02  0:00:02 --:--:-- 2061k
[00:03:36] + mkdir curl-build
[00:03:36] + cd curl-build
[00:03:36] + hide_output ../curl-7.51.0/configure --prefix=/rustroot --with-ssl=/rustroot --disable-sspi --disable-gopher --disable-smtp --disable-smb --disable-imap --disable-pop3 --disable-tftp --disable-telnet --disable-manual --disable-dict --disable-rtsp --disable-ldaps --disable-ldap
[00:03:58] + hide_output make -j10
[00:03:58] + set +x
[00:04:11] shared.sh: line 11:    12 Terminated              bash -c "while true; do sleep 30; echo \$(date) - building ...; done"
[00:04:11] + hide_output make install
---
 94 82.1M   94 77.3M    0     0  3605k      0  0:00:23  0:00:21  0:00:02 3208k
 98 82.1M   98 80.8M    0     0  3605k      0  0:00:23  0:00:22  0:00:01 3451k
100 82.1M  100 82.1M    0     0  3586k      0  0:00:23  0:00:23 --:--:-- 3412k
[00:08:00] + cd gcc-4.8.5
[00:08:00] + sed -i 's|ftp://gcc\.gnu\.org/|http://gcc.gnu.org/|g' ./contrib/download_prerequisites
[00:08:00] --2018-08-21 11:00:30--  http://gcc.gnu.org/pub/gcc/infrastructure/mpfr-2.4.2.tar.bz2
[00:08:00] Resolving gcc.gnu.org... 209.132.180.131
[00:08:00] Connecting to gcc.gnu.org|209.132.180.131|:80... connected.
[00:08:00] HTTP request sent, awaiting response... 200 OK
---
[00:41:50]  ---> 2eab8d65158e
[00:41:50] Step 25/38 : RUN ./build-clang.sh
[00:41:50]  ---> Running in 6ca89c5949a9
[00:41:51] + source shared.sh
[00:41:51] + LLVM=6.0.0
[00:41:51] + mkdir clang
[00:41:51] + cd clang
[00:41:51] + curl https://releases.llvm.org/6.0.0/llvm-6.0.0.src.tar.xz
[00:41:51] + tar xf -
[00:41:51] + xz -d
[00:41:51]                                  Dload  Upload   Total   Spent    Left  Speed
[00:41:54] 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
 34 24.1M   34 8479k    0     0  7191k      0  0:00:03  0:00:01  0:00:02 8378k
 70 24.1M   70 16.8M    0     0  7940k      0  0:00:03  0:00:02  0:00:01 8600k
100 24.1M  100 24.1M    0     0  8111k      0  0:00:03  0:00:03 --:--:-- 8583k
[00:41:54] + cd llvm-6.0.0.src
[00:41:54] + mkdir -p tools/clang
[00:41:54] + curl https://releases.llvm.org/6.0.0/cfe-6.0.0.src.tar.xz
[00:41:54] + xz -d
[00:41:54] + tar xf - -C tools/clang --strip-components=1
[00:41:54]                                  Dload  Upload   Total   Spent    Left  Speed
[00:41:55] 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  1 11.4M    1  147k    0     0  1252k      0  0:00:09 --:--:--  0:00:09 1264k
  1 11.4M    1  147k    0     0  1252k      0  0:00:09 --:--:--  0:00:09 1264k
 65 11.4M   65 7667k    0     0  6853k      0  0:00:01  0:00:01 --:--:-- 6864k
100 11.4M  100 11.4M    0     0  7552k      0  0:00:01  0:00:01 --:--:-- 7559k
[00:41:55] + mkdir ../clang-build
[00:41:55] + cd ../clang-build
[00:41:55] + INC=/rustroot/include
[00:41:55] + INC=/rustroot/include:/rustroot/lib/gcc/x86_64-unknown-linux-gnu/4.8.5/include-fixed
[00:41:55] + INC=/rustroot/include:/rustroot/lib/gcc/x86_64-unknown-linux-gnu/4.8.5/include-fixed:/usr/include
[00:41:55] + hide_output cmake ../llvm-6.0.0.src -DCMAKE_C_COMPILER=/rustroot/bin/gcc -DCMAKE_CXX_COMPILER=/rustroot/bin/g++ -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/rustroot -DLLVM_TARGETS_TO_BUILD=X86 -DC_INCLUDE_DIRS=/rustroot/include:/rustroot/lib/gcc/x86_64-unknown-linux-gnu/4.8.5/include-fixed:/usr/include
[00:42:24] + hide_output make -j10
[00:42:24] + set +x
[00:42:54] Tue Aug 21 11:35:25 UTC 2018 - building ...
[00:43:24] Tue Aug 21 11:35:55 UTC 2018 - building ...
---
[01:33:07] Step 28/38 : RUN ./build-git.sh
[01:33:07]  ---> Running in bb35bcf78f7c
[01:33:07] + source shared.sh
[01:33:07] + tar xzf -
[01:33:07] + curl -L https://www.kernel.org/pub/software/scm/git/git-2.10.0.tar.gz
[01:33:07]                                  Dload  Upload   Total   Spent    Left  Speed
[01:33:08] 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
---
[01:34:08] + set +x
[01:34:09] + hide_output make INSTALL_HDR_PATH=dest headers_install
[01:34:09] + set +x
[01:34:13] shared.sh: line 11:    10 Terminated              bash -c "while true; do sleep 30; echo \$(date) - building ...; done"
[01:34:13] + find dest/include '(' -name .install -o -name ..install.cmd ')' -delete
[01:34:13] + yes
[01:34:13] + yes
[01:34:13] + cp -fr dest/include/asm dest/include/asm-generic dest/include/drm dest/include/linux dest/include/mtd dest/include/rdma dest/include/scsi dest/include/sound dest/include/video dest/include/xen /usr/include
[01:34:13] + rm -rf linux-3.2.84
[01:34:14]  ---> c79326365065
[01:34:14] Removing intermediate container 909e408c9d09
[01:34:14] Step 31/38 : COPY scripts/sccache.sh /scripts/
---
[02:57:06] [RUSTC-TIMING] syntax_pos test:false 3.596
[02:57:06]    Compiling rustc-ap-rustc_errors v218.0.0
[02:57:13] [RUSTC-TIMING] rustc_errors test:false 6.983
[02:58:25] [RUSTC-TIMING] syntax test:false 72.122
The job exceeded the maximum time limit for jobs, and has been terminated.
```

</details><p></p>

[I'm a bot](https://github.com/rust-ops/rust-log-analyzer)! I can only do what humans tell me to, so if this was not helpful or you have suggestions for improvements, please ping or otherwise contact **`@TimNN`**. ([Feature Requests](https://github.com/rust-ops/rust-log-analyzer/issues?q=is%3Aopen+is%3Aissue+label%3Afeature-request))

kennytm commented:
We should add a warning against any non-functional changes to files in `src/ci/docker/scripts/` 😐 

@bors r+

Added back #53329.
Added #53551.

bors commented:
:pushpin: Commit e3887e6c8afbaafe0332e9776199bc89030e227b has been approved by `kennytm`

<!-- @bors r=kennytm e3887e6c8afbaafe0332e9776199bc89030e227b -->

bors commented:
:hourglass: Testing commit e3887e6c8afbaafe0332e9776199bc89030e227b with merge 9f9f2c0095cd683b94adca133f2733aa1f88bb19...

bors commented:
:sunny: Test successful - [status-appveyor](https://ci.appveyor.com/project/rust-lang/rust/build/1.0.8692), [status-travis](https://travis-ci.org/rust-lang/rust/builds/418774098?utm_source=github_status&utm_medium=notification)
Approved by: kennytm
Pushing 9f9f2c0095cd683b94adca133f2733aa1f88bb19 to master...
